src/lib/Files.svelte

```diff
@@ -55,26 +56,43 @@
             })
     })
 
-    function printFile(file: FileEntry, nestLevel: number = 0): string {
+    function createThumbnail(path) {
+        invoke("convert_to_webp", {filePath: path}).catch((err) => {
+            console.error(err)
+        })
+    }
+
+    function printFile(fileEntry: FileEntry, nestLevel: number = 0): string {
         let printString: string = ""
         let nesting = "&nbsp;".repeat(nestLevel * 4)
-        printString += `<p>${nesting}${file.name}</p>`
-        if (file.children) {
-            file.children.forEach((child: FileEntry) => {
-                printString += printFile(child, nestLevel + 1)
-            })
-        }
+        printString += `<p>${nesting}${fileEntry.name}</p>`
+
+        fileEntry.children?.forEach((subDirectory: FileEntry) => {
+            if (subDirectory.name === ".thumbnails" && subDirectory.children) {
+                subDirectory.children.forEach((thumbnail: FileEntry) => {
+                    printString += printFile(thumbnail, nestLevel + 1)
+                })
+            }
+        })
+
         return printString
     }
 
     function changeViewDirectory(fileEntry: FileEntry) {
         viewFiles = []
         if (fileEntry.children) {
             fileEntry.children.forEach((file: FileEntry) => {
-                viewFiles.push({
-                    fileEntry: file,
-                    imageSource: convertFileSrc(file.path)
-                })
+                if (file.name === '.thumbnails') {
+                    file.children?.forEach((subDirectory: FileEntry) => {
+                        viewFiles.push({
+                            fileEntry: subDirectory,
+                            imageSource: convertFileSrc(subDirectory.path)
+                        })
+                    })
+                }
+                if (file.path.endsWith(".tif")) {
+                    createThumbnail(file.path)
+                }
```

Comment:
Disse blir ikke pushet til bruker med en gang. Måtte gå ut og inn av mappen for å vise nyopprettede filer. Kanskje sjekk etter tif før selve push-files-foreach? Eller ignorer webp men push når vi looper over tif? Kan bli lettere å få en fornuftig filending også da.

Reply:
GUIet oppdaterer seg nå automatisk når det skjer endringer i scanner-pathen.